Implementation of output intermediate

zrock-dev opened the pull request:
# Task 138 | Implementation of output intermediate

https://tree.taiga.io/project/joseluis-teran-coffeetime/task/138

## Description
#### What did I do it?
To enable bridge send data to screen
#### How did I do it?
Frame serialization implementation
	- The frames are serialized using flatbuffer library.
	- CoffeeTime is machine generated

implemented a socket client
	- Transmitter takes a message and proceeds to send it through the
	  SocketClient as an array of bytes

Added [flatbuffer](https://flatbuffers.dev) dependency

Generated files will be stored at src/main/java/CoffeeTime.
Updated spotless to ignore CoffeeTime


#### Why did I do it?
To enable game graphics to be rendered in Screen.

## Type of Change

Put an `x` in all the boxes that apply:

- [x] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

- [x] My commit messages are detailed.
- [x] My code follows the code style of this project.
- [x] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests pass.

zrock-dev committed:
feat: implemented a socket client

- Transmitter takes a message and proceeds to send it trough the
  SocketClient as an array of bytes

zrock-dev committed:
feat: Frame serialization implementation

- The frames are serialized using flatbuffer library.
- CoffeeTime is machine generated

zrock-dev committed:
doc: added overall implementation documentation

zrock-dev commented:
The CoffeeTime folder has been generating using [flatcc](https://github.com/dvidelabs/flatcc), and the schema is written in:

https://github.com/Pending-Name-21/miscellany/pull/15

zrock-dev commented:
This implementation doesn't have unit tests because it is required to use `Sprite` and `Sound` classes, but the factories are returning empty objects.

I've created a PR with a refactor to solve this issue.
https://github.com/Pending-Name-21/arquitecture/pull/19

Misax148 reviewed (approved):
Good job

sebastian-barra-jalafund reviewed (approved):
No observations, just add a socket test

zrock-dev committed:
fix: nested flatbuffers

Sprite and Sound tables where being generated during the generation of
the Frame this resulted in an exception. Fixed by serializing them first
and then the Frame.

zrock-dev committed:
chore: added frame testing

To verify if the Transmitter can send a frame.

sebastian-barra-jalafund reviewed (approved):
Test were added, good job

zrock-dev commented:
I forgot to update RenderManager to use the Transmitter.

zrock-dev committed:
chore: updated RenderManager to use Transmitter

Uncommented code related to the implementation and updated constructor
to allow RenderManager have its own transmitter.

zrock-dev committed:
chore: added disconnected server handling

The game will have a frames loss of 60, if these threshold is reached
then a GameException will be produced.

zrock-dev commented:
I have updated this branch with develop. And updated RenderManager to use Transmitter

Misax148 commented on bridge/src/main/java/com/bridge/ipc/SocketClient.java:
```diff
@@ -0,0 +1,68 @@
+ */
+public class SocketClient {
+    public static final Path NAMESPACE =
+            Path.of(System.getProperty("java.io.tmpdir"), "events-socket.sock");
```
The path of the server-socket of Screen is `/tmp/socket_console` and should be the same.
This path I think is of input-intermediate socket

zrock-dev replied:
I've updated it to the suggested path

zrock-dev committed:
chore: updated socket client namespace path

To use the same as screen
